pkg/network/encoding/format.go

```diff
@@ -263,3 +264,16 @@ func unsafeStringSlice(key string) []byte {
 	sh := (*reflect.StringHeader)(unsafe.Pointer(&key))
 	return unsafe.Slice((*byte)(unsafe.Pointer(sh.Data)), len(key))
 }
+
+// formatProtocol converts a single protocols into a protobuf representation of protocol stack.
```

Comment:
Add example for better understanding of the function

Reply:
done